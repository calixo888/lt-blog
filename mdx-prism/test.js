const rehype = require("rehype");
const dedent = require("dedent");

const processHtml = (html, options) => {
  return rehype()
    .data("settings", { fragment: true })
    .use(require("./index"), options)
    .processSync(html)
    .toString();
};

test("finds code and highlights it", () => {
  const result = processHtml(dedent`
        <div>
            <pre><code class="language-jsx{4}">
            class Button extends React.Component {
              state = {
                textColor: slowlyCalculateTextColor(this.props.color)
              };
              render() {
                return (
                  <button
                    className={
                      "Button-" + this.props.color + " Button-text-" + this.state.textColor // ðŸ”´ Stale on color prop updates
                    }
                  >
                    {this.props.children}
                  </button>
                );
              }
            }
            </code></pre>
        </div>
    `);

  expect(result).toMatchSnapshot();
});

module.exports = {
  processHtml
};
