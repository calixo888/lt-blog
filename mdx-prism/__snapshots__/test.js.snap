// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finds code and highlights it 1`] = `
"<div>
    <pre class=\\"language-jsx\\"><code class=\\"language-jsx{4}\\">
    <span class=\\"token keyword\\">class</span> <span class=\\"token class-name\\">Button</span> <span class=\\"token keyword\\">extends</span> <span class=\\"token class-name\\">React<span class=\\"token punctuation\\">.</span>Component</span> <span class=\\"token punctuation\\">{</span>
      state <span class=\\"token operator\\">=</span> <span class=\\"token punctuation\\">{</span>
<div class=\\"mdx-marker\\">        textColor<span class=\\"token punctuation\\">:</span> <span class=\\"token function\\">slowlyCalculateTextColor</span><span class=\\"token punctuation\\">(</span><span class=\\"token keyword\\">this</span><span class=\\"token punctuation\\">.</span>props<span class=\\"token punctuation\\">.</span>color<span class=\\"token punctuation\\">)</span>
</div>      <span class=\\"token punctuation\\">}</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token function\\">render</span><span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span> <span class=\\"token punctuation\\">{</span>
        <span class=\\"token keyword\\">return</span> <span class=\\"token punctuation\\">(</span>
          
            <span class=\\"token punctuation\\">{</span><span class=\\"token keyword\\">this</span><span class=\\"token punctuation\\">.</span>props<span class=\\"token punctuation\\">.</span>children<span class=\\"token punctuation\\">}</span>
          
        <span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>
      <span class=\\"token punctuation\\">}</span>
    <span class=\\"token punctuation\\">}</span>
    </code></pre>
</div>"
`;
